src/pandroid/app/src/main/java/com/panda3ds/pandroid/input/InputHandler.java

```diff
@@ -0,0 +1,116 @@
+package com.panda3ds.pandroid.input;
+
+import android.view.InputDevice;
+import android.view.KeyEvent;
+import android.view.MotionEvent;
+
+import com.panda3ds.pandroid.lang.Function;
+
+import java.util.HashMap;
+
+public class InputHandler {
+    private static Function<InputEvent> eventListener;
+    private static float motionDeadZone = 0.0f;
+
+    private static final int[] gamepadSources = {
+            InputDevice.SOURCE_GAMEPAD,
+            InputDevice.SOURCE_JOYSTICK
+    };
+
+    private static final int[] validSources = {
+            InputDevice.SOURCE_GAMEPAD,
+            InputDevice.SOURCE_JOYSTICK,
+            InputDevice.SOURCE_DPAD,
+            InputDevice.SOURCE_KEYBOARD
+    };
+
+    private static final HashMap<String, Float> motionDownEvents = new HashMap<>();
+
+    private static boolean containsSource(int[] sources, int sourceMasked) {
+        for (int source : sources) {
```

Comment:
What is this...?

Reply:
Android stores input source (input type), in a masked int, i'll storing all valids sources in `validSources`, this function is for check `sourceMasked` is in a valid list, i maked a function because i need separate in for check source is a gamepad, for prevent confilit check function:  `processKeyEvent`